src/inspector/inspector.service.ts

```diff
@@ -42,14 +42,15 @@ export class InspectorService implements OnModuleInit {
         const { slotToWrite, stateRoot, slotNumber } = await this.waitForNextFinalizedSlot(nextFinalizedSlot);
         if (slotToWrite > 0) {
           const res = await this.dataProcessor.processAndWriteFinalizedData(slotToWrite, stateRoot, slotNumber);
+          let possibleHighRewardValidators: string[] = [];
           if (res?.lidoIDs.length) {
             const headEpoch = await this.calculateHeadEpoch();
-            const possibleHighRewardValidators = await this.dataProcessor.getPossibleHighRewardValidatorIndexes(res.lidoIDs, headEpoch);
-            await this.statsProcessor.calculateLidoStats(slotToWrite, possibleHighRewardValidators);
-            await this.statsProcessor.calculateOtherStats(res.otherCounts);
-            await this.statsProcessor.finalizeAppIterate(slotToWrite);
-            await this.criticalAlertService.sendCriticalAlerts(slotToWrite);
+            possibleHighRewardValidators = await this.dataProcessor.getPossibleHighRewardValidatorIndexes(res.lidoIDs, headEpoch);
           }
+          await this.statsProcessor.calculateLidoStats(slotToWrite, possibleHighRewardValidators);
+          await this.statsProcessor.calculateOtherStats(res.otherCounts);
```

Comment:
Is here a possibility to add these 2 calls to `Promise.all`?

Reply:
Not sure if this makes sense, but I'll think about it in another task (some optimization, for example)